### DISCUSSION TITLE
Need suggestion on Ribo data

### DISCUSSION THREAD

**Discussion by Saurabh-plantist** [zhpn1024/ribotish#38] [General]
I need your valuable suggestions regarding Ribotish.

We are currently working with Arabidopsis and have generated sequencing data from three Ribo-Seq libraries to evaluate the effectiveness of our protocols. Each library produced an average of 35 million reads. After removing adapters, low-quality reads, and ribosomal RNA reads, we are left with an average of 30 million high-quality reads per library. I mapped the clean reads to the reference genome using STAR alignment with the following command:
"STAR --runThreadN 4 --genomeDir Star_genome/TAIR10_index --readFilesIn bnon_LUrRNA_7.1.fastq non_LUrRNA_7.2.fastq --outFileNamePrefix Star_out/For_ribotish/Ribo_7 --outSAMtype BAM SortedByCoordinate --outSAMattributes NM NH MD --alignEndsType EndToEnd".
The BAM files generated were indexed using samtools. Later, the ribotish both quality and predict commands run as follows:
" ribotish quality -b Ribo_7Aligned.sortedByCoord.out.bam -g Arabidopsis_thaliana_ChrC.gtf -l 20,35 --nom0 -p 4"
"ribotish predict -b Ribo_7Aligned.sortedByCoord.out.bam -g Arabidopsis_thaliana_ChrC.gtf -f TAIR10_clean.fa -o 7_pred.txt -p 4"
Subject: Clarification Needed on P-site Offset Results
I have some concerns regarding the P-site offset results, which do not appear satisfactory (please refer to the attached files). Additionally, while running the parameter "predict", the *.txt file is empty, while the *all.txt file shows some results (again, see the attached files). I need to figure out how to interpret these findings.

I also used Ribocode, which did not yield any results and suggested that the sequencing quality might need to be improved. I would appreciate your thoughts on this matter.

Am I on the right track to proceed with our more extensive experiments using the same strategy? Do we need more coverage or depth in the sequencing, or do you believe modifications to the experimental design are necessary?

I'm looking forward to hearing your expert comments.

[7_pred_framebest.txt](https://github.com/user-attachments/files/17708153/7_pred_framebest.txt)
[7_pred_framebest_all.txt](https://github.com/user-attachments/files/17708154/7_pred_framebest_all.txt)
[Ribo_7Aligned.sortedByCoord.out_qual.pdf](https://github.com/user-attachments/files/17708155/Ribo_7Aligned.sortedByCoord.out_qual.pdf)

Thank you!
Kind regards
Saurabh

**Comment by zhpn1024** [comment]
Dear Saurabh,
The upper part of the quality figure showed that the available CDS reads were only 70k level. I suggest checking these points:
Did the annotation file include all CDSs in Arabidopsis? Do not use lncRNA only annotation in the quality step.
Where did the 30 million high-quality reads mapped to? Were most of the reads longer than 34 nt? If most reads were longer, it might be some experimental problem. We can change the '-l 20,35' option to cover longer reads.
Try not using '--nom0' in the quality module.

Best,
Peng
